File: examples/E06-heranca-simples/expected/lib/animais.dart
abstract class Animal {
  Animal();

  Animal.syntaxBridgeCopyOf(Animal other) {}

  String falar();

  String apresentar() {
    return 'Eu digo: ' + falar();
  }
}

class Cachorro extends Animal {
  Cachorro();

  Cachorro.syntaxBridgeCopyOf(Cachorro other)
    : super.syntaxBridgeCopyOf(other) {}

  @override
  String falar() {
    return 'Au au';
  }
}

class Gato extends Animal {
  Gato();

  Gato.syntaxBridgeCopyOf(Gato other) : super.syntaxBridgeCopyOf(other) {}

  @override
  String falar() {
    return 'Miau';
  }
}

String apresentarAnimal(Animal animal) {
  return animal.apresentar();
}

String testarCachorro() {
  Cachorro c = Cachorro();
  return apresentarAnimal(c);
}

String testarGato() {
  Gato g = Gato();
  return apresentarAnimal(g);
}

File: examples/E06-heranca-simples/input/src/animais.cpp
#include "animais.hpp"

Animal::~Animal() {}

std::string Animal::apresentar() const {
    return "Eu digo: " + falar();
}

std::string Cachorro::falar() const {
    return "Au au";
}

std::string Gato::falar() const {
    return "Miau";
}

std::string apresentarAnimal(const Animal& animal) {
    return animal.apresentar();
}

std::string testarCachorro() {
    Cachorro c;
    return apresentarAnimal(c);
}

std::string testarGato() {
    Gato g;
    return apresentarAnimal(g);
}
